### PR TITLE
improve contribution guide

### PR DESCRIPTION
**Pull request by pswu11**
the purposes of this pull request

- add basic structure to the contribution guide 
- add a table of content for users to look for relevant info
- add a section for maintainers -> an attempt to suggest something that help core engineers to organize issues, feel free to comment on it

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,25 +1,23 @@
...
+- 🧭 [Navigation](#navigation)
+- 🔧 [Environment setup](#environment-setup)
+- ✅ [Before you submit](#before-you-submit)
+- 📢 [For maintainers](#for-maintainers) 
 
-## Contribute a PR
+# Bugs and features
 
-### Requirements
+Whether it's a bug report or feature request, you're welcome to raise **[issues](https://github.com/questdb/questdb/issues)** using respective templates.
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:14]:
```suggestion
Whether it's a bug report or feature request, you're welcome to raise an **[issue](https://github.com/questdb/questdb/issues)** using the respective template. 
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,25 +1,23 @@
...
-  ([Homebrew](https://github.com/Homebrew/brew)) or
-  [from the jar](https://maven.apache.org/install.html) for any OS)
-- Node.js 12 / npm 6 (to manage your Node.js versions we recommend
-  [nvm](https://github.com/nvm-sh/nvm) for OSX/Linux/windows WSL, and
-  [nvm-windows](https://github.com/coreybutler/nvm-windows) for Windows) -
-  OPTIONAL
-- C-compiler, CMake - to contribute to C libraries - OPTIONAL
+If you're not sure whether you should raise an issue, you can also join our community **[Slack channel](https://slack.questdb.io/)** and post your questions there. 
+
+We aim to respond to your issues and questions soonest. If you wish to receive a faster response, we recommend you to always describe your steps and provide information about your environment in your bug reports. And if you're proposing a new feature, it'll help us to evaluate the priority if you explain why you need a specific feature.
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:18]:
```suggestion
We aim to respond to your issues and questions soonest. If you wish to receive a faster response, we recommend you always describe your steps and provide information about your environment in the bug reports. And if you're proposing a new feature, it'll help us to evaluate the priority if you explain why you need it.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -39,9 +37,31 @@ deep diving into the code base.
 Compiled binaries (for C libraries and Windows service wrapper) are committed to
 git to make build of development process Java-centric and simplified.
 
-# Local setup
+## Find suitable Sissues
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:40]:
```suggestion
## Finding suitable issues
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -39,9 +37,31 @@ deep diving into the code base.
 Compiled binaries (for C libraries and Windows service wrapper) are committed to
 git to make build of development process Java-centric and simplified.
 
-# Local setup
+## Find suitable Sissues 
+
+Our maintainers label issues with relevant categories so you can use that to search for certain type of issues you'd like to work on. 
+If you don't know where to start, try search for issues labeled with `good first issue` or `help wanted`.
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:43]: nit: might be a good idea to place links with filtered issues here.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -39,9 +37,31 @@ deep diving into the code base.
...
+# Environment setup
+
+## Requirements
+
+- Operating system - **x86-64**: Windows, Linux, FreeBSD and OSX
+- Java 11 64-bit
+- Maven 3 (from your package manager on Linux / OSX
+  ([Homebrew](https://github.com/Homebrew/brew)) or
+  [from the jar](https://maven.apache.org/install.html) for any OS)
+- Node.js 12 / npm 6 (to manage your Node.js versions we recommend
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:56]: nit: I think it's already Node.js 16 and npm 7. @insmac should know for sure.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -61,7 +81,7 @@ Windows
 set JAVA_HOME="c:\path\to\java directory"
 ```
 
-## Compiling Java and frontend code
+### Compiling Java and frontend code
 
 Compiling the database + the web console can be done with:
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:86]:
```suggestion
You can compile the database and build the web console with the following command:
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -77,7 +97,7 @@ java -p core/target/questdb-<version>-SNAPSHOT.jar -m io.questdb/io.questdb.Serv
 
 The web console will available at [localhost:9000](http://localhost:9000).
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:98]:
```suggestion
The web console will be available at [localhost:9000](http://localhost:9000).
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -129,35 +149,37 @@ The build will copy artifacts as follows:
...
+# Before you submit
+
+## Testing
 
 We have a lot of unit tests, most of which are of "integration" type, e.g. test
 starts a server, interacts with it and asserts the outcome. We expect all
-contributors to submit PRs with tests. Please reach out to us via slack if you
-uncertain on how to test, or you think existing test is inadequate and should be
+contributors to submit PRs with tests. Please reach out to us via slack if you're
+uncertain on how to test, or you think an existing test is inadequate and should be
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:159]: nit: "could be improved" sounds nicer than "is inadequate and should be removed".
```suggestion
uncertain on how to test, or you think an existing test could be improved.
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -129,35 +149,37 @@ The build will copy artifacts as follows:
...
 removed.
 
-# Dependencies
+## Dependencies
 
 QuestDB does not have dependencies. This may sound unorthodox but in reality we
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:164]:
```suggestion
QuestDB does not have Java dependencies. This may sound unorthodox but in reality we
```

**Review by puzpuzpuz — COMMENTED** [review]
Please run `prettier -w CONTRIBUTING.md` to reformat the file properly. I can help with that if needed.

**Comment by puzpuzpuz** [comment]
I really like the improvements. 👍

File: CONTRIBUTING.md (hunk 5 above)
Comment (insmac) [reply CONTRIBUTING.md:67]: In my view our requirements should be:
`Node.js 16 / npm 8` (I would make `8` required as it's shipped with npm with the exception of FreeBSD which has these decoupled)

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -39,15 +45,40 @@ deep diving into the code base.
...
+- C-compiler, CMake - to contribute to C libraries - _OPTIONAL_
+
+## Local environment
 
-## Setup Java and JAVA_HOME
+### Setting up Java and `JAVA_HOME`
 
-JAVA*HOME is required by Maven. It is possible to have multiple version of Java
-on the same platform. Please set up JAVA_HOME to point to Java 11. Other
+Java Home is required by Maven. It is possible to have multiple version of Java
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:77]:
```suggestion
`JAVA_HOME` is required by Maven. It is possible to have multiple versions of Java
```

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -39,15 +45,40 @@ deep diving into the code base.
...
+on the same platform. Please set up `JAVA_HOME` to point to Java 11. Other
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:78]:
```suggestion
on the same platform. Please set up `JAVA_HOME` to point to Java 11 or later. Other
```

File: CONTRIBUTING.md [hunk 12]
```diff
@@ -39,15 +45,40 @@ deep diving into the code base.
...
+or [`help wanted`](https://github.com/questdb/questdb/labels/Help%20wanted). If
+you wish to understand how our maintainers work together, you can refer to
+[this section](#for-maintainers).
+
+# Environment setup
+
+## Requirements
+
+- Operating system - **x86-64**: Windows, Linux, FreeBSD and OSX
+- Java 11 64-bit
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:63]:
```suggestion
- Java 11 64-bit or later
```

File: CONTRIBUTING.md [hunk 13]
```diff
@@ -75,12 +107,13 @@ You can then run QuestDB with:
...
-The web console will available at [localhost:9000](http://localhost:9000).
+The web console will be available at [localhost:9000](http://localhost:9000).
 
-## Compiling C-libraries
+### Compiling C-libraries
 
 C-libraries will have to be compiled for each platform separately. Cmake will
-also need JAVA_HOME to be set. The following commands will compile on Linux/OSX.
+also need `JAVA_HOME` to be set. The following commands will compile on
+Linux/OSX.
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:116]:
```suggestion
Linux/OS X.
```

File: CONTRIBUTING.md [hunk 14]
```diff
@@ -129,35 +162,37 @@ The build will copy artifacts as follows:
 ui -> core/src/main/resources/io/questdb/site/public.zip
 ```
 
-# Testing
+# Before you submit
+
+## Testing
 
 We have a lot of unit tests, most of which are of "integration" type, e.g. test
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:169]:
```suggestion
We have a lot of tests, most of which are of "integration" type, e.g. test
```

File: CONTRIBUTING.md [hunk 15]
```diff
@@ -129,35 +162,37 @@ The build will copy artifacts as follows:
...
 starts a server, interacts with it and asserts the outcome. We expect all
-contributors to submit PRs with tests. Please reach out to us via slack if you
-uncertain on how to test, or you think existing test is inadequate and should be
-removed.
+contributors to submit PRs with tests. Please reach out to us via slack if
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:171]:
```suggestion
contributors to submit PRs with tests. Please reach out to us via Slack if
```

File: CONTRIBUTING.md [hunk 16]
```diff
@@ -196,18 +231,18 @@ mvn clean package -DskipTests -P build-web-console
 
 ### How can I easily manage the `Node.js` installation?
 
-You can use [nvm](https://github.com/nvm-sh/nvm) for OSX/Linux/windows WSL, and
-[nvm-windows](https://github.com/coreybutler/nvm-windows) for Windows. To have
-multiple active versions.
+You can use [`nvm`](https://github.com/nvm-sh/nvm) for OSX/Linux/windows WSL,
```
Comment (puzpuzpuz) [inline CONTRIBUTING.md:234]:
```suggestion
You can use [`nvm`](https://github.com/nvm-sh/nvm) for OS X/Linux/Windows WSL,
```

**Comment by puzpuzpuz** [comment]
Left some minor comments.

**Comment by marregui** [comment]
Excellent work!!
I was wondering if we have a policy on IDEs, do we support all?.

**Comment by puzpuzpuz** [comment]
> I was wondering if we have a policy on IDEs, do we support all?.

I think, we should support any IDE as long as it supports Maven projects (and that means all Java IDEs).

**Review by puzpuzpuz — APPROVED** [review]
Great job!

**Pull request merged**